### PR TITLE
Added default policies document

### PR DESCRIPTION
**Pull request by adrianhall**
This is the ordering from @JeffreyRichter

### CODE REVIEW

File: docs/design/DefaultPolicies.mdk [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## Default HTTP Pipeline Policies {#general-default-policies}
+
+Most client libraries will use the HTTP pipeline for their request/response transport to the service.
+
+~ Must
+Implement the following policies in the HTTP pipline:
```
Comment (JonathanGiles) [inline docs/design/DefaultPolicies.mdk]:
```suggestion
Implement the following policies in the HTTP pipeline:
```

Comment (adrianhall) [reply docs/design/DefaultPolicies.mdk]: Good catch!

File: docs/design/DefaultPolicies.mdk [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+~ Must
+Implement the following policies in the HTTP pipeline:
+- [Telemetry](#general-telemetry)
+- Unique Request ID
+- [Retry](#general-retry)
+- Credentials
+- Response downloader
+- Distributed tracing
+- [Logging](#general-logging)
```
Comment (bterlson) [inline docs/design/DefaultPolicies.mdk]: I think proxy support can be implemented as a pipeline policy, and so should be.

**Comment by JeffreyRichter** [comment]
I think it just be part of the transport policy and not a specific additional policy?


  *   Jeffrey Richter (watch Architecting Distributed Cloud Apps<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fazure.microsoft.com%2Ftraining%2Fdistributed-cloud-apps%2F&data=02%7C01%7Cjeffreyr%40microsoft.com%7C948cf3366acf45f6ef8d08d62e3b3b00%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636747229263047172&sdata=sg%2FtzHWpscWQDr7foOSsaNrKDP5vNSHE3b72HJG8swU%3D&reserved=0>)

**Comment by bterlson** [comment]
HTTP proxies can be used by munging the request in a similar way to other policies (namely, add Host header) and seems orthogonal to transport? Also, since #294 suggests a few configurations to check, being able to swap out the policy to use different configuration options from the system seem like a useful thing, and having to swap out the entire transport seems overkill.

File: docs/design/DefaultPolicies.mdk (hunk 2 above)
Comment (adrianhall) [reply docs/design/DefaultPolicies.mdk]: Agreed - added.

File: docs/design/DefaultPolicies.mdk [hunk 3]
```diff
@@ -0,0 +1,17 @@
+## Default HTTP Pipeline Policies {#general-default-policies}
+
+Most client libraries will use the HTTP pipeline for their request/response transport to the service.
+
+~ Must
+Implement the following policies in the HTTP pipeline:
```
Comment (johanste) [inline docs/design/DefaultPolicies.mdk]: For Python, we also have a NetworkTracingPolicy, RedirectPolicy and UserAgent policy.

Comment (pakrym) [reply docs/design/DefaultPolicies.mdk]: Isn't telemetry the same as User-agent policy?

Comment (johanste) [reply docs/design/DefaultPolicies.mdk]: Ah, yes, probably.

Comment (JeffreyRichter) [reply docs/design/DefaultPolicies.mdk]: Yes it is. Some languages (JavaScript) doesn't have control over the user-agent so "telemetry" must be done another way this is why the policy is named after its purpose and not its implementation.

File: docs/design/DefaultPolicies.mdk (hunk 2 above)
Comment (pakrym) [reply docs/design/DefaultPolicies.mdk]: The proxy policy might be non-trivial to implement, do we really want to pull the battle-tested feature from HTTP client libraries and reimplement and own them?

**Comment by pakrym** [comment]
> Also, since #294 suggests a few configurations to check, being able to swap out the policy to use different configuration options from the system seem like a useful thing, and having to swap out the entire transport seems overkill.

Why not add a way to configure default transports instead of pulling things from them to make them configurable?

For example, on windows, WinHTTP based HTTP client would support system-wide proxy settings by default so do we reimplement the feature or do we disable it to have our own proxy policy implementation?

I would argue that proxy is a transport implementation detail similar to SSL and the benefits of having it as a policy are not clear while disadvantages are numerous.

**Comment by adrianhall** [comment]
We discussed proxy and whether to make it a policy or transport option extensively in the last architectural review board.  Whether the proxy policy is included or not is determined by whether we architecturally prefer it as a policy or transport.  The arch. board decided to make it a policy.

File: docs/design/DefaultPolicies.mdk (hunk 3 above)
Comment (adrianhall) [reply docs/design/DefaultPolicies.mdk]: So, do we have both Telemetry + UserAgent policies, or just Telemetry policy?

**Comment by pakrym** [comment]
> Whether the proxy policy is included or not is determined by whether we architecturally prefer it as a policy or transport

I don't think I fully understand.

> The arch. board decided to make it a policy.

It looked like there was still a conversation happening about that in this PR that's why I asked.

**Comment by JeffreyRichter** [comment]
My recollection from yesterday was that we decided on a separate policy for proxy because we might want to log the HTTP 407 challenge and so the log policy has to be between the transport policy and the proxy policy. Also, if 2FA or some other behavior is required to be approved by the proxy, a separate policy gives us a place to accommodate that.

**Pull request merged**